### PR TITLE
Feat/32 input component

### PR DESCRIPTION
**Pull request by Trent1900**
### Descriptions
UI only. 
including storybook and testing.
add the search bar for desktop and ipad, and search Icon for mobile phone. 
- Link to UI design (if possible):
image

### Provide screenshots or videos that the function is working.
image

### Provide screenshots of no warning, or error in the console.
image

- [x] All unit tests passing and coverage of testing is more than 90% unless special needs.
- [x] No merge conflict with the `main` branch, make sure your code is up to date with `main`.
- [x] Check PR title is aligned with your ticket.
- [x] Check your app will run without crashes.
- [x] Check if there are hard-coded values.
- [x] No `//eslint-disable-next-line no-unused-vars`, unless otherwise specified.
- [x] Check you are using `camelCase && PascalCase` correctly.
- [x] Remove unused comments.

### CODE REVIEW

File: src/components/Input/Search.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          Search for music, album and artist
+        </label>
+        <div className="pointer-events-none absolute inset-y-0 left-0 hidden items-center pl-3 sm:flex">
+          <AiOutlineSearch className="h-4 w-4 text-light-400"></AiOutlineSearch>
+        </div>
+        <input
+          id="searchInput"
+          placeholder="Search for music, album and artist"
+          className="hidden h-9 w-full items-center rounded-full border border-light-100 bg-black/75 py-1.5 px-3 
+          pl-8 text-xl text-light-400 focus:border-light-400 focus:outline-none focus:placeholder:text-light-400 sm:flex"
```
Comment (Double-Twelve) [inline src/components/Input/Search.tsx:32]: 1. should be `py-2` `pl-9` `focus:placeholder:text-transparent`
2. should remove `items-center`
3. should add `placeholder:text-light-100` `hover:border-light-400` `hover:placeholder:text-light-400`

File: src/components/Input/Search.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
+import React, { useCallback, useState } from "react";
```
[the remark this answers is not in the document]
Comment (Trent1900) [reply src/components/Input/Search.tsx:1]: thanks

File: src/components/Input/Search.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import { AiOutlineSearch } from "react-icons/ai";
+
+interface ISearchProps {
+  onSubmit?: React.FormEventHandler<HTMLFormElement> | (() => void);
+  onClick?: React.MouseEventHandler<SVGElement> | (() => void);
+}
+
+const Search = <T extends ISearchProps>({ onSubmit, onClick }: T) => {
```
[the remark this answers is not in the document]
Comment (Trent1900) [reply src/components/Input/Search.tsx:9]: try to use generic way to define, it is not necessary, will change back to normal type.

File: src/components/Input/Search.tsx (hunk 1 above)
Comment (Trent1900) [reply src/components/Input/Search.tsx:32]: thanks for reviewing

File: src/components/Input/Search.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+
+  const handleSearch = onSubmit;
+  const expendMaskLayer = onClick;
+
+  const handleInput = (e: React.ChangeEvent<HTMLInputElement>) => {
+    setInputValue(e.target.value);
+  };
+
+  return (
+    <>
```
[the remark this answers is not in the document]
Comment (Trent1900) [reply src/components/Input/Search.tsx:20]: thanks

**Pull request closed without merging**